lib/components/viewers/live-stop-times.tsx

```diff
@@ -144,36 +169,72 @@ class LiveStopTimes extends Component<Props, State> {
       return stopTimeComparator(stopTimesA[0], stopTimesB[0])
     }
 
+    const routeTimes = Object.values(stopTimesByPattern)
+      .filter(({ times }) => times.length !== 0)
+      .sort(patternComparator)
+      .map((route) => {
+        const sortedTimes = route.times
+          .concat()
+          ?.sort(stopTimeComparator)
+          // Only show times within 24 hours of next arrival time
+          .filter((time: any, i: number, array: Array<any>) => {
+            const firstDepartureTime =
+              array[0].serviceDay + array[0].realtimeDeparture
+            const departureTime = time.serviceDay + time.realtimeDeparture
+            return i === 0 || (departureTime - firstDepartureTime) / 3600 < 24
+          })
+        const { serviceDay } = sortedTimes[0]
+        return {
+          ...route,
+          day: serviceDay,
+          times: sortedTimes
+        }
+      })
+      .filter(({ pattern, route }) =>
+        routeIsValid(route, getRouteIdForPattern(pattern))
```

Comment:
Any way we can move that previous filter into this filter step? Just so we don't loop through the array twice

Reply:
I can move the `routeIsValid` filter into the `times !== 0` filter no problem! The 2nd service day filter below is a bit tricker, because it relies on the `sort(stopTimeComparator)` which  in turn needs the `times !== 0` filter to avoid throwing an error. There could definitely be a better way to do this that I'm missing!